src/components/search-input/search-input.module.scss

```diff
@@ -0,0 +1,46 @@
+.form {
+    position: relative;
+    left: 1em;
+    width: 16em;
+
+    @media (max-width: 768px) {
+        width: 10em;
+    }
+}
+
+.input {
+    width: 100%;
+    padding: 0 1.5em;
+    border: none;
+    background: none;
+    font-size: 0.9em;
+
+    &::placeholder {
+        font-size: inherit;
+    }
+
+    &:focus-visible {
+        box-shadow: none;
+    }
+}
+
+.searchIcon {
+    position: absolute;
+    top: 50%;
+    left: 0;
+    transform: translateY(-50%);
+    display: inline-block;
+}
```

Comment:
Nitpick, but I think this would be easier to achieve (and maybe easier for users to style) if implemented using flexbox instead of absolute positioning.

Something like this: https://jsfiddle.net/1ynr2ect/

Reply:
Oh, and wrapping everything with a label so the input receives focus if you click on the search icon or the "x" icon would be a nice touch.

Reply:
wrapped with label. used flex.